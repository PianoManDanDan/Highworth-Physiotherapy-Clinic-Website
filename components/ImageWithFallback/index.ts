export * from './ImageWithFallback';
export * from './ImageWithFallbackProps';
