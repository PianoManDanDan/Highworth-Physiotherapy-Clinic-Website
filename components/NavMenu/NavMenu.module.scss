@use 'styles/mixins/media-query' as *;

@keyframes example {
  0% {
    display: inline;
    opacity: 1;
    transform: rotate(0deg);
  };

  5% {
    transform: rotate(-10deg);
  };
  95% {
    transform: rotate(190deg);
  };

  99% {
    // display: none;
    width: 24px;
    height: 24px;
    opacity: 0;
    transform: rotate(180deg);
  };
  100% {
    width: 0;
    height: 0;
  }
};

.menu {
  // display: none;
  @include media-query(small) {
    display: flex;
    width: 80%;
  };
  @include media-query(medium) {
    width: 65%;
  };
  @include media-query(large) {
    width: 50%;
  };

  &__open {
    background: red;
  };

  &Icon {
    @include media-query(small) {
      display: none;
    };

    &__hide {
      animation-name: example;
      animation-duration: 1s;
      animation-timing-function: ease-in-out;
      animation-iteration-count: 1;
      animation-direction: alternate;
      animation-fill-mode: both;
    };
  };
};