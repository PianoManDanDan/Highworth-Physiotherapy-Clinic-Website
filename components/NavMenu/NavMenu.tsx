import React, { useState } from 'react';
import cn from 'classnames';
import { Menu, X } from 'react-feather';
import styles from './NavMenu.module.scss';

export const NavMenu: React.FC<{}> = ({ children }) => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <nav
      className={cn({
        [styles.menu]: true,
        [styles['menu--open']]: isOpen,
      })}
    >
      <Menu
        className={cn({
          [styles.menu__icon]: true,
          [styles['menu__icon--hide']]: isOpen,
        })}
        onClick={() => setIsOpen(true)}
      />
      <X
        className={cn({
          [styles.menu__icon]: true,
          [styles['menu__icon--hide']]: !isOpen,
        })}
        onClick={() => setIsOpen(false)}
      />
      {children}
    </nav>
  );
};

export default NavMenu;
