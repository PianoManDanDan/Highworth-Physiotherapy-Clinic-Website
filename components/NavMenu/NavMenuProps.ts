export type NavMenuProps = {
    /** Child nodes */
    children: React.ReactNode | React.ReactNode[];
  };