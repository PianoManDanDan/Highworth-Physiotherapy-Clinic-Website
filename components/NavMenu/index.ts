export * from './NavMenu';
export * from './subcomponents';