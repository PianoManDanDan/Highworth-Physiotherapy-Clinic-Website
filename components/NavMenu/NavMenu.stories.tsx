import React from 'react';
import { StoryObj, Meta } from '@storybook/react';
import { NavMenu, NavMenuItem } from 'components/NavMenu';

export default {
  component: NavMenu,
  subcomponents: { NavMenuItem }
} as Meta;

export const desktopView: StoryObj<typeof NavMenu> = {
  render: () => (
    <NavMenu>
      <NavMenuItem text='Test 1' href='/test-1' />
      <NavMenuItem text='Test 2' href='/test-2' />
      <NavMenuItem text='Test 3' href='/test-3' />
    </NavMenu>
  ),
}

export const mobileView: StoryObj<typeof NavMenu> = {
  render: () => (
    <NavMenu>
      <NavMenuItem text='Test 1' href='/test-1' />
      <NavMenuItem text='Test 2' href='/test-2' />
      <NavMenuItem text='Test 3' href='/test-3' />
    </NavMenu>
  ),
  parameters: {
    viewport: {
      defaultViewport: 'mobile1',
    },
  },
}
