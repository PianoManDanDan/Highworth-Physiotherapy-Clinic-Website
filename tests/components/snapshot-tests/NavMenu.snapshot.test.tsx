import React from 'react';
import { render } from '@testing-library/react';
import { NavMenu, NavMenuItem } from '../../../components/NavMenu';

describe('NavMenu Component snapshot test', () => {
  describe('When passed NavMenuItem components as children', () => {
    test('it renders the NavMenu element with all of its\' children', () => {
      // Arrange
      const { asFragment } = render(
        <NavMenu>
          <NavMenuItem text='Child NavMenuItem 1' href='/child-1-href' />
          <NavMenuItem text='Child NavMenuItem 2' href='/child-2-href' />
          <NavMenuItem text='Child NavMenuItem 3' href='/child-3-href' />
        </NavMenu>
      );

      // Act
      const renderedNavMenu = asFragment();

      // Assert
      expect(renderedNavMenu).toMatchSnapshot();
    });
  });
  describe('When passed ordinary html elements as children', () => {
    test('it renders the NavMenu element with all of its\' children', () => {
      // Arrange
      const { asFragment } = render(
        <NavMenu>
          <p>Child 1</p>
          <div>Child 2</div>
          <p>Child 3</p>
        </NavMenu>
      );
  
      // Act
      const renderedNavMenu = asFragment();
  
      // Assert
      expect(renderedNavMenu).toMatchSnapshot();
    });
  });
});
