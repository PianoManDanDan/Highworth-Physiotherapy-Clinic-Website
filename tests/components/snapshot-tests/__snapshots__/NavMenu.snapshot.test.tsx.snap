// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavMenu Component snapshot test When passed NavMenuItem components as children it renders the NavMenu element with all of its' children 1`] = `
<DocumentFragment>
  <nav>
    <a
      href="/child-1-href"
    >
      Child NavMenuItem 1
    </a>
    <a
      href="/child-2-href"
    >
      Child NavMenuItem 2
    </a>
    <a
      href="/child-3-href"
    >
      Child NavMenuItem 3
    </a>
  </nav>
</DocumentFragment>
`;

exports[`NavMenu Component snapshot test When passed ordinary html elements as children it renders the NavMenu element with all of its' children 1`] = `
<DocumentFragment>
  <nav>
    <p>
      Child 1
    </p>
    <div>
      Child 2
    </div>
    <p>
      Child 3
    </p>
  </nav>
</DocumentFragment>
`;
